app/web_conn.go

```diff
@@ -94,6 +94,16 @@ func (c *WebConn) SetSession(v *model.Session) {
 	c.session.Store(v)
 }
 
+func (c *WebConn) Pump() {
+	ch := make(chan struct{}, 1)
+	go func() {
```

Comment:
Why leave this go routine here? Can’t we wrap it?

Reply:
My goal isn’t to eliminate goroutines. It’s just to make sure goroutines have well defined lifetimes and don’t outlive their dependencies. In the same way App now guarantees that there are no dependant goroutines executing when Shutdown returns, WebConn now guarantees that there are no dependant goroutines executing when Pump returns. I think it’s better for objects to be accountable for their own goroutines.